src/orion/algo/space.py

```diff
@@ -181,6 +187,21 @@ def name(self, value):
         else:
             raise TypeError("Dimension's name must be either string or None. "
                             "Provided: {}, of type: {}".format(value, type(value)))
+    
+    @property
+    def default_value(self):
+        return self._default_value
+
+    @default_value.setter
+    def default_value(self, value):
```

Comment:
I'm not sure we should have a public setter. Dimensions object are basically immutable objects, with the exception of their random number generator in ```self.prior```.